code/modules/reagents/chemistry/reagents/misc.dm

```diff
@@ -219,13 +221,35 @@
 		var/datum/effect_system/smoke_spread/bad/BS = new
 		BS.set_up(1, 0, T)
 		BS.start()
-		if(!QDELETED(old_holder))
-			old_holder.add_reagent("ash", round(volume * 0.5))
+		if(!QDELETED(old_holder) && reagent_after_burning)
+			old_holder.add_reagent(reagent_after_burning, round(volume * 0.5))
 
 /datum/reagent/oil/reaction_turf(turf/T, volume)
 	if(volume >= 3 && !isspaceturf(T) && !locate(/obj/effect/decal/cleanable/blood/oil) in T)
 		new /obj/effect/decal/cleanable/blood/oil(T)
 
+/datum/reagent/oil/cooking
+	name = "Cooking Oil"
+	id = "cooking_oil"
+	description = "You should probably pour this down the sink, where it belongs."
+	color = "#fbba16"
+	taste_description = "old french fries"
+	reagent_after_burning = "cooking_oil_inert"
+
+/datum/reagent/oil/cooking/reaction_turf(turf/T, volume)
+	if(volume >= 3 && !isspaceturf(T) && !locate(/obj/effect/decal/cleanable/blood/oil/cooking) in T)
```

Comment:
I think the locate in T might need to be wrapped in parentheses since you're using `!`
```suggestion
	if(volume >= 3 && !isspaceturf(T) && !(locate(/obj/effect/decal/cleanable/blood/oil/cooking) in T))
```

Reply:
This was a copy-paste from the regular oil `reaction_turf` proc so I'll just update it there too